src/action.c

```diff
@@ -144,11 +169,31 @@ action_str_from_arg(struct action_arg *arg)
 	return ((struct action_arg_str *)arg)->value;
 }
 
+static char *
+str_from_key(struct action *action, const char *key, char *default_value)
+{
+	assert(key);
+	struct action_arg *arg;
+	wl_list_for_each(arg, &action->args, link) {
+		if (!strcmp(key, arg->key)) {
+			return action_str_from_arg(arg);
+		}
```

Comment:
`strcasecmp()` to handle silly dev mistakes in the future? By adding the default value those mistakes might be hidden during testing so I feel like leaving a bit more space for error correction in this case could be appropriate.

As alternative we could also define that all internal action arg keys have to be lowercase, that should likely be mentioned in the code style and function docs for `action_arg_add_xxx()` / `xxx_from_key()`.

Reply:
Good point. Have gone for strcasecmp